### ISSUE TITLE
error report using approx on list

### ISSUE THREAD

**Issue by oxlade39** [pytest-dev/pytest#4650]
When using `pytest.approx` to compare large lists of floats I'd like an error report much more like the helpful output of when just comparing lists (without approx). Instead I get something like the following:

```
AssertionError: test_lists
[1.1, 5, 3.3, 4.4] != approx([1.1 ± 1.0e-1...-12, 4.4 ± 1.0e-12])

Expected :approx([1.1 ± 1.0e-1...-12, 4.4 ± 1.0e-12])
Actual   :[1.1, 5, 3.3, 4.4]
```
With a large list it's impossible to tell where the miss-match is.

Example test-case:
```python
def test_lists():
    list_one = [1.1, 2.2, 3.3, 4.4]
    list_two = [1.1, 5, 3.3, 4.4]

    assert list_one == pytest.approx(list_two)
```
- [/ ] Include a detailed description of the bug or suggestion
- [x] `pip list` of the virtual environment you are using
- [/] pytest and operating system versions
platform linux -- Python 3.6.7, pytest-4.1.1, py-1.7.0, pluggy-0.8.1 -- 
- [/] Minimal example if possible

**Comment by nicoddemus** [comment]
Thanks for the report @oxlade39!

cc @kalekundert @tadeu

**Comment by oxlade39** [comment]
Exactly. Although I'm not sure if that's the best output one could hope for but at least it's consistent. 

Anything to make it easier to see where the actual miss-match is would be hugely beneficial

**Comment by Zac-HD** [comment]
Closing as resolved: implemented in #8429 (tracking #8335) — `pytest.approx` failures now show mismatched-element counts, max absolute/relative differences, and an index/obtained/expected diff table; verified with this issue's example on current main.
